pkg/analyzer/nodes/node.go

```diff
@@ -103,22 +104,45 @@ func extractFilesFromDirectory(root string, matcher matcher.PrefixMatcher) (*fil
 	m := metrics.FileExtractionMetrics{}
 	// TODO(ROX-13771): Use `range matcher.GetCommonPrefixDirs()` again after fixing.
 	for _, dir := range []string{"etc/", "usr/share/rpm", "var/lib/rpm", "usr/share/buildinfo"} {
-		if err := n.addFiles(filepath.FromSlash(dir), matcher, &m); err != nil {
+		if err := n.addFiles(ctx, filepath.FromSlash(dir), matcher, &m); err != nil {
 			return nil, errors.Wrapf(err, "failed to match filesMap at %q (at %q)", dir, n.root)
 		}
 	}
 	m.Emit()
 	return n, nil
 }
 
+func walkDirWithContext(ctx context.Context, dir string, fn fs.WalkDirFunc) error {
+	errC := make(chan error)
+	go func(ctx context.Context) {
+		defer close(errC)
+		errC <- filepath.WalkDir(dir, func(path string, info fs.DirEntry, err error) error {
+			select {
+			case <-ctx.Done():
+				return ctx.Err()
+			default:
+				return fn(path, info, err)
+			}
+		})
+	}(ctx)
+
+	// Wait for the WalkDir to complete or for the context to be cancelled.
+	select {
+	case <-ctx.Done():
+		return ctx.Err()
```

Comment:
It might be worth revisiting the [error log triggered by this](https://github.com/stackrox/scanner/pull/1118/files#diff-8757df52309cf15678c26e87d2de2e18ba065a1826da17959e169e4a323b9b0aR108). It mentions the files map could not be matched, but for timeouts, the timeout cancellation next to that message might be confusing.

Reply:
Would sth like this be okay ?

```
node_test.go:469: Error: failed to extract files from "etc/" at root "/var/folders/.../001": operation took longer than expected and was interrupted: context deadline exceeded
```